### PR TITLE
feat(kcp): introduce ability to recursively lookup extensions within kcp

### PR DESCRIPTION
**Pull request by aaronschweig**

### CODE REVIEW

File: pkg/subroutines/extensions.go [hunk 1]
```diff
@@ -157,34 +164,69 @@ func collectExtensions(ctx context.Context, cl client.Client, lookupNamespace st
...
+			}
+
+			clusterName := ws.Annotations[logicalcluster.AnnotationKey]
+
+			parentCtx := kontext.WithCluster(ctx, logicalcluster.Name(clusterName))
+
+			parentAccount := v1alpha1.Account{}
+			err = cl.Get(parentCtx, types.NamespacedName{
+				Name:      ws.Annotations[v1alpha1.NamespaceAccountOwnerLabel],
+				Namespace: ns,
```
Comment (nexus49) [inline pkg/subroutines/extensions.go:191]: So this only works when all accounts are located in the same namespace within the workspaces right?

File: pkg/subroutines/extensions.go [hunk 2]
```diff
@@ -157,34 +164,69 @@ func collectExtensions(ctx context.Context, cl client.Client, lookupNamespace st
...
+				continue
+			}
+
+			clusterName := ws.Annotations[logicalcluster.AnnotationKey]
+
+			parentCtx := kontext.WithCluster(ctx, logicalcluster.Name(clusterName))
+
+			parentAccount := v1alpha1.Account{}
+			err = cl.Get(parentCtx, types.NamespacedName{
+				Name:      ws.Annotations[v1alpha1.NamespaceAccountOwnerLabel],
```
Comment (nexus49) [inline pkg/subroutines/extensions.go:190]: and for now the assumption would be, that this annotation made its way somehow to the workspace and we can add this logic in a later PR or configuration

File: pkg/subroutines/extensions.go (hunk 1 above)
Comment (aaronschweig) [reply pkg/subroutines/extensions.go:191]: Yes, an alternative would be to also include the namespace information in the workspace annotations. Could also be a valid option, as we need to add some information to the annotations anyway.

should I go ahead and add it?

File: pkg/subroutines/extensions.go (hunk 2 above)
Comment (aaronschweig) [reply pkg/subroutines/extensions.go:190]: yes, probably this annotation could find its way through the extension mechanism of the organization level accounts. There we could define a metadata section with this (and potentially the namespace annotation) information based on the account which created it.

File: pkg/subroutines/extensions.go (hunk 1 above)
Comment (nexus49) [reply pkg/subroutines/extensions.go:191]: i'd suggest experiment with this now and add the namespace in a follow up PR

**Review by nexus49 — APPROVED** [review]

**Pull request merged**